src/controller/CHIPDevice.h

```diff
@@ -204,6 +225,14 @@ class DLL_EXPORT Device
 
     SecureSessionMgr<Transport::UDP> * mSessionManager;
 
+    /* Track all outstanding response callbacks for this device. The callbacks are
+       registered when a command is sent to the device, to get notified with the results. */
+    Callback::CallbackDeque mResponses;
+
+    /* Track all outstanding callbacks for attribute reports from this device. The callbacks are
+       registered when the user requests attribute reporting for device attributes. */
+    Callback::CallbackDeque mReports;
```

Comment:
these are registered persistently?

Reply:
Persistently across reboots/restarts? If so, no, not yet. But it can be done in future.